tasks/winbuildscripts/Generate-Chocolatey-Package.ps1

```diff
@@ -15,12 +15,12 @@ $env:chocolateyUseWindowsCompression = 'true'; Invoke-Expression ((New-Object Sy
 pip3 install -r requirements.txt
 
 $outputDirectory = "c:\mnt\build-out"
-$rawAgentVersion = (inv agent.version)
+$rawAgentVersion = (inv agent.version --url-safe --major-version 7)
 $copyright = "Datadog {0}" -f (Get-Date).Year
 
 $releasePattern = "(\d+\.\d+\.\d+)"
 $releaseCandidatePattern = "(\d+\.\d+\.\d+)-rc\.(\d+)"
-$develPattern = "(\d+\.\d+\.\d+)-devel\+git\.\d+\.(.+)"
+$develPattern = "(\d+\.\d+\.\d+)-devel\.git\.\d+\.(.+)"
```

Comment:
Can you share the case that led to this change for the pattern?

Reply:
I needed to use the `--url-safe` agent version (`inv agent.version --url-safe --major-version 7`) to [get the correct name of the MSI file from the gitlab artifacts](https://github.com/DataDog/datadog-agent/blob/f96890f29981bd9821eabe04d5d4c55ad36a7c2b/tasks/winbuildscripts/Generate-Chocolatey-Package.ps1#L71). The [--url-safe version replaces `+` with `.` in the devel pattern](https://github.com/DataDog/datadog-agent/blob/f7b3b58049deb3a0bbc7b14046744464f3ded9ec/tasks/libs/releasing/version.py#L305-L315), which required updating the regex from `+` to `.`

I made sure that the change doesn't affect the `$agentVersion` variable used for the Chocolatey package. The values for group 1 and group 2 remain the same in the develPattern matching logic. 

The groups are determined by the parenthesis:
`$develPattern` = "**(\d+\.\d+\.\d+)**-devel\.git\.\d+\.**(.+)**"
group 1: `(\d+\.\d+\.\d+)`
group 2: `(.+)`